app/controllers/users_controller.rb

```diff
@@ -15,15 +13,17 @@ class UsersController < ApplicationController
     if current_user
       redirect_to({id: current_user.id}, alert: t('common.errors.not_found_error'))
     else
-      redirect_to root_path, alert: t('common.errors.not_authorized')
+      redirect_to root_path, alert: t('common.errors.not_found_error')
     end
   end
 
   def index
     @users = User.all.paginate(page: params[:page], per_page: per_page_param)
+    authorize @users
   end
```

Comment:
Not your fault, but we don't have an index route or view. Hence, we can probably drop this here.